### PR TITLE
Feat/enhancing worker logs

### PR DESCRIPTION
**Pull request by muhammad-ali-e**
## What

- Enhanced logs for tool executions by including container id/name.
- Added optional environment variables to improve developer experience.

## Why

- To improve observability and develper exeperiances

## How

- Improved log formats
- Added optional variables for specifying log level and automatic removal of containers.

## Can this PR break any existing features. If yes please list of possible items. If no please exaplin why. (PS: Admins do not merge the PR without this section filled)

- No. This PR aims to change log formats and introduce optional environment variables.

## Database Migrations

-  No

## Env Config

-  Optional ENVs
  - **LOG_LEVEL:** _Specifies the log level for the worker._
  - **REMOVE_CONTAINER_ON_EXIT:** _Determines whether to remove the tool container after execution._
    -  This will help to check logs in the container even completed the execution

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

- Verify worker logs during execution.
- Please ensure tools are running properly.

## Screenshots
Screenshot from 2024-04-30 13-44-36

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: worker/src/unstract/worker/worker.py [hunk 1]
```diff
@@ -112,40 +116,41 @@ def stream_logs(
...
-            else:
+            if isinstance(log_dict, dict):
                 return log_dict
-        except json.JSONDecodeError as e:
-            logger.warn(
-                f"Received invalid JSON log message: {log_message} \n " f"error: {e}"
-            )
-        return None
+        except json.JSONDecodeError:
+            return None
```
Comment (johnyrahul) [inline worker/src/unstract/worker/worker.py:140]: why is it returning none?

**Review by johnyrahul — APPROVED** [review]
LGTM. 
Raised a  single query

File: worker/src/unstract/worker/worker.py [hunk 2]
```diff
@@ -112,40 +116,41 @@ def stream_logs(
...
+
+        Returns:
+            Optional[dict[str, Any]]: json message
+        """
         try:
             log_dict = json.loads(log_message)
-            if not isinstance(log_dict, dict):
-                logger.error(f"Received invalid log: {log_dict}")
-            else:
+            if isinstance(log_dict, dict):
```
Comment (chandrasekharan-zipstack) [inline worker/src/unstract/worker/worker.py:137]: Do you think this check is necessary @muhammad-ali-e ?

**Review by chandrasekharan-zipstack — APPROVED** [review]

**Review by harini-venkataraman — APPROVED** [review]

File: worker/src/unstract/worker/worker.py (hunk 1 above)
Comment (muhammad-ali-e) [reply worker/src/unstract/worker/worker.py:140]: The purpose of this function is  to get the execution logs to push it to pubsub for FE logs. If we getting  JSONDecodeError we don't raise the error or logging it, instead we return None. Also we logging the all mesages at stream_logs methods.

File: worker/src/unstract/worker/worker.py (hunk 2 above)
Comment (muhammad-ali-e) [reply worker/src/unstract/worker/worker.py:137]: This to confirm the return object is a dict. We exepecting it for FE logs to publish to pubsub

**Pull request merged**